packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js

```diff
@@ -2165,6 +2165,26 @@ QUnit.module('custom mask maskRules', moduleConfig, () => {
             value: ' y'
         });
     });
+
+    QUnit.test('caret moving correctly if maskChar set to empty string', function(assert) {
+        const $textEditor = $('#texteditor').dxTextEditor({
+            mask: '00',
+            maskChar: ''
+        });
+
+        const $input = $textEditor.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+
+        caretWorkaround($input);
+
+        const keyboard = keyboardMock($input, true);
+        keyboard.caret(0);
+
+        keyboard
+            .type('1')
+            .type('2');
+
```

Comment:
Let's first of all test exactly that caret position is correct. Incorrect accepted input value is more like a side effect here.